org.openhab.binding.zigbee/src/main/java/org/openhab/binding/zigbee/internal/converter/config/ZclReportingConfig.java

```diff
@@ -0,0 +1,179 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.zigbee.internal.converter.config;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Objects;
+
+import org.eclipse.jdt.annotation.NonNull;
+import org.eclipse.smarthome.config.core.ConfigDescriptionParameter;
+import org.eclipse.smarthome.config.core.ConfigDescriptionParameter.Type;
+import org.eclipse.smarthome.config.core.ConfigDescriptionParameterBuilder;
+import org.eclipse.smarthome.config.core.Configuration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.zsmartsystems.zigbee.zcl.ZclCluster;
+
+/**
+ * Configuration handler for reporting of digital attributes. This should be applied for each channel - ie if a channel
```

Comment:
What is the reason for fixing the same reporting configuration for all attributes? Is it simply for having a simpler structure of the configuration? Or is it so rare that it is sensible to configure different reporting options for different attributes used by one channel?

Reply:
Attribute reporting is the same for all attributes. Well, to be more specific, there are two types - digital and analogue, and this handler will handle both types, so it's not exactly the same for all attributes.

Maybe I misunderstand your point somehow?

Reply:
My question relates to the JavaDoc added here, which says:

> This should be applied for each channel - ie if a channel uses multiple attributes, then they should all be configured the same.

Why is it required to configure all attributes used within a channel the same? Assume a channel uses two digital attributes `A` and `B` - why not permit to configure reporting for those two attributes differently?

Reply:
I'm not 100% sure why I wrote this, but I think this was referring to a channel where you might get the data from multiple attributes (ie it's not just a case of reading a single attribute).

There are some examples (eg pressure sensor) but I think maybe it's confusing (especially as I don't recall why I wrote this!).

Reply:
ok, thanks Chris!